### PR TITLE
prow.sh: deploy canary snapshot-controller in canary jobs

### PR DESCRIPTION
**Pull request by pohly**
https://k8s-testgrid.appspot.com/sig-storage-csi-ci#canary-on-master was using
the snapshot-controller from the YAML file in the master branch, which is the
latest stable release. What needs to be tested instead is the canary image to
detect regressions before a release.

### CODE REVIEW

**Comment by pohly** [comment]
Here's some example output from a local test:
```
csi-driver-host-path$ CSI_PROW_BUILD_JOB=false CSI_PROW_TESTS=serial-alpha CSI_PROW_DRIVER_CANARY=canary CSI_PROW_KUBERNETES_VERSION=1.21.0 ./.prow.sh
...
Installing snapshot CRDs from https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd
customresourcedefinition.apiextensions.k8s.io/volumesnapshotclasses.snapshot.storage.k8s.io created
customresourcedefinition.apiextensions.k8s.io/volumesnapshots.snapshot.storage.k8s.io created
customresourcedefinition.apiextensions.k8s.io/volumesnapshotcontents.snapshot.storage.k8s.io created
No resources found
No resources found in default namespace.
No resources found
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
serviceaccount/snapshot-controller created
clusterrole.rbac.authorization.k8s.io/snapshot-controller-runner created
clusterrolebinding.rbac.authorization.k8s.io/snapshot-controller-role created
role.rbac.authorization.k8s.io/snapshot-controller-leaderelection created
rolebinding.rbac.authorization.k8s.io/snapshot-controller-leaderelection created
NAME                       ROLE                                     AGE
snapshot-controller-role   ClusterRole/snapshot-controller-runner   0s
Deploying snapshot-controller from https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml with canary images.
29c29
<         image: k8s.gcr.io/sig-storage/snapshot-controller:v4.2.0
---
>         image: gcr.io/k8s-staging-sig-storage/snapshot-controller:canary
deployment.apps/snapshot-controller created
12:46:57 waiting for snapshot controller deployment to complete, attempt #0
```

File: prow.sh [hunk 1]
```diff
@@ -832,8 +832,19 @@ install_snapshot_controller() {
               echo "$modified"
               exit 1
           fi
-	  echo "kubectl apply -f ${SNAPSHOT_CONTROLLER_YAML}(modified)"
       done
+  elif [ "${CSI_PROW_DRIVER_CANARY}" = "canary" ]; then
```
Comment (xing-yang) [inline prow.sh:836]: @pohly, should we also check if CSI_PROW_KUBERNETES_VERSION is the latest?
```
if [ "${CSI_PROW_KUBERNETES_VERSION}" = "latest" ] || [ "${CSI_PROW_DRIVER_CANARY}" = "canary" ]; then
```

Comment (xing-yang) [reply prow.sh:836]: I see the following in the logs:
```
external-snapshotter version tag: CSI_SNAPSHOTTER_VERSION=master
using           image: k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.3
```
csi-snapshotter:v3.0.3 is from `https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.19`.
So we are testing csi-snapshotter:v3.0.3 with snapshot-controller:v4.2.0 (latest stable version).

Comment (pohly) [reply prow.sh:836]: The question still is: do you want to test the latest stable snapshot-controller in the "on-kubernetes-master" job or the latest canary?

**Pull request merged**